### PR TITLE
Allow null value. boolean <-> Boolean error

### PR DESCRIPTION
**Pull request by KostyaSha**
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/605)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/core/DefaultDockerClientConfig.java [hunk 1]
```diff
@@ -319,7 +321,7 @@ public String toString() {
         private String apiVersion, registryUsername, registryPassword, registryEmail, registryUrl, dockerConfig,
                 dockerCertPath;
 
-        private boolean dockerTlsVerify;
+        private Boolean dockerTlsVerify;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/DefaultDockerClientConfig.java]: either setter should use boolean. If `null` - then docker-java default.

**Comment by KostyaSha** [comment]
@marcuslinke please review

**Comment by marcuslinke** [comment]
@KostyaSha LGTM!

**Comment by marcuslinke** [comment]
@KostyaSha Oh, a test is failing! Haven' seen this before. So it doesn't LGTM ;)

**Comment by KostyaSha** [comment]
@marcuslinke sorry, fixed

File: src/test/java/com/github/dockerjava/core/DefaultDockerClientConfigTest.java [hunk 2]
```diff
@@ -180,25 +180,25 @@ public void withDockerTlsVerify() throws Exception {
...
-        assertThat(field.getBoolean(builder), is(false));
+        assertThat((Boolean) field.get(builder), is(false));
 
         builder.withDockerTlsVerify("false");
-        assertThat(field.getBoolean(builder), is(false));
+        assertThat((Boolean) field.get(builder), is(false));
 
         builder.withDockerTlsVerify("FALSE");
-        assertThat(field.getBoolean(builder), is(false));
+        assertThat((Boolean) field.get(builder), is(false));
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/DefaultDockerClientConfigTest.java]: btw, why not move setters directly into class, then getters would work?

Comment (marcuslinke) [inline src/test/java/com/github/dockerjava/core/DefaultDockerClientConfigTest.java]: Sorry, don't get it...

Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/DefaultDockerClientConfigTest.java]: field is object, getBoolean() works only with primitive

**Pull request merged**